pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -38,35 +39,23 @@ nParams=${#params[@]}
 while (( "$n" < "$nParams" )); do
     p=${params[n]}
     p2=${params[n+1]:-} # handle `p` being last one
-    if [ "$p" = -c ]; then
-        dontLink=1
-    elif [ "$p" = -S ]; then
-        dontLink=1
-    elif [ "$p" = -E ]; then
-        dontLink=1
-    elif [ "$p" = -E ]; then
-        dontLink=1
-    elif [ "$p" = -M ]; then
-        dontLink=1
-    elif [ "$p" = -MM ]; then
-        dontLink=1
-    elif [[ "$p" = -x && "$p2" = *-header ]]; then
-        dontLink=1
-    elif [[ "$p" = -x && "$p2" = c++* && "$isCxx" = 0 ]]; then
-        isCxx=1
-    elif [ "$p" = -nostdlib ]; then
-        cxxLibrary=0
-    elif [ "$p" = -nostdinc ]; then
-        cInclude=0
-        cxxInclude=0
-    elif [ "$p" = -nostdinc++ ]; then
-        cxxInclude=0
-    elif [[ "$p" != -?* ]]; then
-        # A dash alone signifies standard input; it is not a flag
-        nonFlagArgs=1
-    elif [ "$p" = -cc1 ]; then
-        cc1=1
-    fi
+
+    case "$p" in
+        -[cSEM] | -MM) dontLink=1 ;;
+        -cc1) cc1=1 ;;
+        -nostdinc) cInclude=0 cxxInclude=0 ;;
+        -nostdinc++) cxxInclude=0 ;;
+        -nostdlib) cxxLibrary=0 ;;
+        -target|--target=*) needsTarget=0;;
```

Comment:
Is this automatically passed through?

Reply:
Yep params comes from https://github.com/NixOS/nixpkgs/pull/176152/files#diff-5b214d4b19e3f1e9146a43659772fb2e7a5830badc02d53087ea62f5f5860b93R34
and params is maybe reset in the NIX_ENFORCE_PURITY if branch, but that basically filters out any -I/-L/-isystem args that `badPath` doesn't like from params. similar happens for march/mtune=native filtering.

So in the end ${params[@]} ends up with the params provided by the caller possibly without any impure paths or mtune/march flags